src/config.js

```diff
@@ -0,0 +1,1602 @@
+  // ```
+  //
+  // * `keyPath` The {String} name of the key.
+  // * `value` The value of the setting. Passing `undefined` will revert the
+  //   setting to the default value.
+  // * `options` (optional) {Object}
+  //   * `scopeSelector` (optional) {String}. eg. '.source.ruby'
+  //     See [the scopes docs](http://flight-manual.atom.io/behind-atom/sections/scoped-settings-scopes-and-scope-descriptors/)
+  //     for more information.
+  //   * `source` (optional) {String} The name of a file with which the setting
+  //     is associated. Defaults to the user's config file.
+  //
+  // Returns a {Boolean}
+  // * `true` if the value was set.
+  // * `false` if the value was not able to be coerced to the type specified in the setting's schema.
+  set () {
+    let [keyPath, value, options = {}] = Array.from(arguments)
+
+    if (!this.settingsLoaded) {
+      this.pendingOperations.push(() => this.set(keyPath, value, options))
+    }
+
+    const scopeSelector = options.scopeSelector
+    let source = options.source
+    const shouldSave = options.save != null ? options.save : true
+
+    if (source && !scopeSelector) {
+      throw new Error("::set with a 'source' and no 'sourceSelector' is not yet implemented!")
+    }
+
+    if (source == null) { source = this.getUserConfigPath() }
+
+    if (value !== undefined) {
+      try {
+        value = this.makeValueConformToSchema(keyPath, value)
+      } catch (e) {
+        return false
+      }
+    }
+
+    if (scopeSelector != null) {
+      this.setRawScopedValue(keyPath, value, source, scopeSelector)
+    } else {
+      this.setRawValue(keyPath, value)
+    }
+
+    if ((source === this.getUserConfigPath()) && shouldSave && !this.configFileHasErrors && this.settingsLoaded) {
+      this.requestSave()
+    }
+    return true
+  }
+
+  // Essential: Restore the setting at `keyPath` to its default value.
+  //
+  // * `keyPath` The {String} name of the key.
+  // * `options` (optional) {Object}
+  //   * `scopeSelector` (optional) {String}. See {::set}
+  //   * `source` (optional) {String}. See {::set}
+  unset (keyPath, options) {
+    if (!this.settingsLoaded) {
+      this.pendingOperations.push(() => this.unset(keyPath, options))
+    }
+
+    let {scopeSelector, source} = options != null ? options : {}
+    if (source == null) { source = this.getUserConfigPath() }
+
+    if (scopeSelector != null) {
+      if (keyPath != null) {
+        let settings = this.scopedSettingsStore.propertiesForSourceAndSelector(source, scopeSelector)
+        if (getValueAtKeyPath(settings, keyPath) != null) {
+          this.scopedSettingsStore.removePropertiesForSourceAndSelector(source, scopeSelector)
+          setValueAtKeyPath(settings, keyPath, undefined)
+          settings = withoutEmptyObjects(settings)
+          if (settings != null) { this.set(null, settings, {scopeSelector, source, priority: this.priorityForSource(source)}) }
```

Comment:
Seems like a long line, this `if` is begging for breaking it down to separate lines